### PR TITLE
eliminate misleading flag parsing error

### LINKED ISSUE #1321: Unhelpful message on incorrect weave launch option
Example:

    # weave launch -p
    The weave container has died. Consult the logs with 'docker logs weave' for further details.
    # docker logs weave
    flag provided but not defined: -p
    See '/home/weave/weaver --help'.

The path `/home/weave/weaver` is internal to the container, so the user would find it hard to follow this advice.

### PR DESCRIPTION
**Pull request by rade**
The standard mflags error reporting produces an error like
```
flag provided but not defined: -p
See '/home/weave/weaver --help'.
```
on stderr. This is misleading since the program name is container local, so not something the user actually invoked. Furthermore, what the user *can* invoke doesn't have a `--help` option.

With this change the error becomes
```
FATA: 2015/08/18 10:56:23.635119 Incorrect usage: flag provided but not defined: -p
```

There are a few issues with this:

- we no longer get a usage message
- we no longer get deprecation warnings logged

Fixes #1321. Alas I am not at all happy with this. Wasted several hours on this already. Somebody else please have a go.

### CODE REVIEW

**Comment by awh** [comment]
````
vagrant@vagrant-ubuntu-vivid-64:~/weave$ ./weave launch -p
The weave container has died. Consult the logs with 'docker logs weave' for further details.
vagrant@vagrant-ubuntu-vivid-64:~/weave$ docker logs weave
flag provided but not defined: -p
See 'weave help'.
````
Feel free to squash if you're happy with the approach.

**Comment by rade** [comment]
@awh see what you make of this one. I've left the old commits in and reverted them, just in case we change our mind once more. If not, I'll squash them away.

**Comment by awh** [comment]
That seems entirely reasonable to me, given the lack of `flags` support for overriding the executable name in sane fashion...

**Comment by awh** [comment]
The only other option appears to be vendoring `mflags`, which strikes me as wholly unappealing :unamused:
